flink-runtime/src/main/java/org/apache/flink/runtime/operators/sort/CircularQueues.java

```diff
@@ -75,6 +75,9 @@ public void send(StageRunner.SortStage stage, CircularElement<E> element) {
 
 	@Override
 	public void sendResult(MutableObjectIterator<E> result) {
+		if (iteratorFuture.isDone()) {
```

Comment:
I think this and the changes in `MergeIterator`, `SpillingThread`, and `NormalizedKeySorter` should be in a separate cleanup/refactoring commit.

Reply:
Actually these changes are unnecessary, I added them for helping me with debugging and forgot to remove them.